test/config/test_config.py

```diff
@@ -116,93 +114,88 @@ def test_interpolation_cross_section(self):
             }
         )
         self.cfg._from_cp(self.cp)
-        self.assertFalse(self.cfg.get('general', 'ignore-case'))
-        self.assertFalse(self.cfg.get('network', 'do-not-track'))
+        assert not self.cfg.get('general', 'ignore-case')
+        assert not self.cfg.get('network', 'do-not-track')
 
     def test_invalid_interpolation(self):
         """Test an invalid interpolation."""
         self.cp.read_dict({'general': {'ignore-case': '${foo}'}})
         self.cfg._from_cp(self.cp)
-        with self.assertRaises(configparser.InterpolationError):
+        with pytest.raises(configparser.InterpolationError):
             self.cfg._validate_all()
 
     def test_invalid_interpolation_syntax(self):
         """Test an invalid interpolation syntax."""
         self.cp.read_dict({'general': {'ignore-case': '${'}})
-        with self.assertRaises(configexc.InterpolationSyntaxError):
+        with pytest.raises(configexc.InterpolationSyntaxError):
             self.cfg._from_cp(self.cp)
 
     def test_invalid_section(self):
         """Test an invalid section."""
         self.cp.read_dict({'foo': {'bar': 'baz'}})
-        with self.assertRaises(configexc.NoSectionError):
+        with pytest.raises(configexc.NoSectionError):
             self.cfg._from_cp(self.cp)
 
     def test_invalid_option(self):
         """Test an invalid option."""
         self.cp.read_dict({'general': {'bar': 'baz'}})
-        with self.assertRaises(configexc.NoOptionError):
+        with pytest.raises(configexc.NoOptionError):
             self.cfg._from_cp(self.cp)
 
     def test_invalid_section_relaxed(self):
         """Test an invalid section with relaxed=True."""
         self.cp.read_dict({'foo': {'bar': 'baz'}})
         self.cfg._from_cp(self.cp, relaxed=True)
-        with self.assertRaises(configexc.NoSectionError):
+        with pytest.raises(configexc.NoSectionError):
             self.cfg.get('foo', 'bar')  # pylint: disable=bad-config-call
 
     def test_invalid_option_relaxed(self):
         """Test an invalid option with relaxed=True."""
         self.cp.read_dict({'general': {'bar': 'baz'}})
         self.cfg._from_cp(self.cp, relaxed=True)
-        with self.assertRaises(configexc.NoOptionError):
+        with pytest.raises(configexc.NoOptionError):
             self.cfg.get('general', 'bar')  # pylint: disable=bad-config-call
 
 
-class DefaultConfigTests(unittest.TestCase):
+class TestDefaultConfig:
 
     """Test validating of the default config."""
 
+    @pytest.mark.xfail(reason='fails if run with other tests in this module, '
+                              'succeeds if executed alone (The-Compiler, help)'
```

Comment:
Congrats, you spotted a bug! :laughing: 

`config.ConfigManager` initializes its default values from `configdata.DATA`. That's a dict `{str: Section}`, and a section contains the values.

Now since `DATA` is a global, when multiple `ConfigManager`s are used, they actually *share* their sections, which means the change from `test_invalid_interpolation` gets carried over, tested and fails.

I'll need to check what the best fix for this is - probably using a factory function in `configdata` instead of a global `DATA`.